src/support/SupportSummary.tsx

```diff
@@ -0,0 +1,78 @@
+import ReactJson from '@microlink/react-json-view';
+import React, { useEffect, useState } from 'react';
+import { FullyQualifiedResourceId } from 'src/libs/ajax/SamResources';
+import colors from 'src/libs/colors';
+import { reportError } from 'src/libs/error';
+import { ResourceTypeSummaryProps } from 'src/support/SupportResourceType';
+
+export const SupportSummary = (props: ResourceTypeSummaryProps) => {
+  const [summaryInfo, setSummaryInfo] = useState<object>();
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const resourceId = props.fqResourceId.resourceId;
+  const fqResourceId = props.fqResourceId;
+
+  async function loadSupportSummary(id: FullyQualifiedResourceId) {
+    if (props.loadSupportSummaryFn) {
+      return props.loadSupportSummaryFn(id);
+    }
+    // throw error here because this component should not be used without a summary function
+    throw new Error('No support summary function provided');
+  }
+
+  function clear() {
+    setErrorMessage('');
+    setSummaryInfo(undefined);
+  }
+
+  useEffect(() => {
+    const loadSummary = async () => {
+      clear();
+      if (resourceId) {
+        try {
+          setSummaryInfo(await loadSupportSummary(fqResourceId));
+        } catch (e) {
+          if (e instanceof Response && e.status === 404) {
+            setErrorMessage(`${props.displayName} not found`);
+          } else if (e instanceof Response && e.status === 403) {
+            setErrorMessage(
+              `You do not have permission to view ${props.displayName} summary information or are not on VPN`
+            );
+          } else {
+            await reportError('Error loading group summary', e);
+          }
+        }
+      }
+    };
+
+    loadSummary();
+    // eslist complains about loadSupportSummary being a dependency, but adding it causes an infinite loop
```

Comment:
Ignoring dependencies often leads to bugs. Is there a way to get rid of this?

In this case, the infinite loop is caused because `loadSupportSummary` is created on each render, so it has always changed. This could be resolved by either memoizing it with a `useCallback` hook or (preferably) inlining it within the effect function.

Reply:
Yes, this makes me very queasy.